adam-core/src/main/scala/org/bdgenomics/adam/rdd/settheory/Closest.scala

```diff
@@ -0,0 +1,190 @@
+package org.bdgenomics.adam.rdd.settheory
+
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models.ReferenceRegion
+import org.bdgenomics.adam.rdd.ManualRegionPartitioner
+import org.bdgenomics.utils.interval.array.IntervalArray
+import scala.reflect.ClassTag
+
+/**
+ * A trait describing closest implementations that are based on sort-merge.
+ *
+ * @tparam T The type of the left records.
+ * @tparam U The type of the right records.
+ * @tparam RT The resulting type of the left after the join.
+ * @tparam RU The resulting type of the right after the join.
+ */
+sealed abstract class Closest[T: ClassTag, U: ClassTag, RT, RU]
```

Comment:
Closest doesn't necessarily overlap, and is performed between two `RDD`s. In your proposed architecture, in order to be truly exhaustive we'd end up joining each record on the left with all records with the same `referenceName` on the right.